src/core/streams/stream_message_renderer.js

```diff
@@ -34,3 +39,55 @@ function getPermanentElementMapForFragment(fragment) {
 
   return permanentElementMap
 }
+
+async function withAutofocusFromFragment(fragment, callback) {
+  const generatedID = `turbo-stream-autofocus-${Date.now()}`
+  const turboStreams = fragment.querySelectorAll("turbo-stream")
+  const elementWithAutofocus = firstAutofocusableElementInStreams(turboStreams)
+  let willAutofocus = generatedID
```

Comment:
Should `willAutofocus` be falsy if there's no `elementWithAutofocus`? We could move the assignment to the next conditional.